server/src/http/instance_data.rs

```diff
@@ -27,65 +26,43 @@ pub enum InstanceDataSending {
-            EdgeMode::Edge(edge_args) => {
-                let identifier = instance_data.identifier.clone();
-                edge_args
-                    .tokens
-                    .first()
-                    .map(|token| {
-                        let client_meta_information = ClientMetaInformation {
-                            app_name: args.app_name,
-                            instance_id: identifier.clone(),
-                            connection_id: identifier,
-                        };
-                        let http_client = new_reqwest_client(
-                            edge_args.skip_ssl_verification,
-                            edge_args.client_identity.clone(),
-                            edge_args.upstream_certificate_file.clone(),
-                            Duration::seconds(edge_args.upstream_request_timeout),
-                            Duration::seconds(edge_args.upstream_socket_timeout),
-                            client_meta_information.clone(),
-                        )
-                        .expect(
-                            "Could not construct reqwest client for posting observability data",
-                        );
-                        let unleash_client = Url::parse(&edge_args.upstream_url.clone())
-                            .map(|url| {
-                                UnleashClient::from_url(
-                                    url,
-                                    args.auth_headers
-                                        .upstream_auth_header
-                                        .clone()
-                                        .unwrap_or("Authorization".to_string()),
-                                    http_client,
-                                    client_meta_information.clone(),
-                                )
-                            })
-                            .map(|c| {
-                                c.with_custom_client_headers(
-                                    edge_args.custom_client_headers.clone(),
-                                )
-                            })
-                            .map(Arc::new)
-                            .map_err(|_| {
-                                EdgeError::InvalidServerUrl(edge_args.upstream_url.clone())
-                            })
-                            .expect("Could not construct UnleashClient");
-                        let instance_data_sender = InstanceDataSender {
-                            unleash_client,
-                            token: token.clone(),
-                            base_path: args.http.base_path.clone(),
-                            registry,
-                        };
-                        InstanceDataSending::SendInstanceData(instance_data_sender)
-                    })
-                    .map(Ok)
-                    .unwrap_or(Ok(InstanceDataSending::SendNothing))
-            }
+            EdgeMode::Edge(edge_args) => edge_args
```

Comment:
We now pass in a reqwest Client and a ClientMetaInformation rather than constructing them inside the function.  Other than that this has been left more or less unchanged. It needs a bit of love probably but in the next PR